Polishing documentation (#22)
uniqorg/eslint-extensions: apps/website/docs/plugins/intro.md

sidebar_position: 1
---

> # what are eslint plugins?

Reviewer: ```suggestion
# What are eslint plugins?
```

> `eslint` plugins provide third-party to extend eslint.

Reviewer: ```suggestion
`eslint` plugins are third-party extensions for eslint.
```
or: `eslint` plugins provide third-party -something (rules?/conventions?/configs?)- to extend eslint.

Before using the plugin, you have to install it.

> Because this website focuses on opinionated conventions, here some references for more general understanding you can read these docs:

Reviewer: ```suggestion
Because our docs focus on opinionated conventions, for more general understanding here are some references/docs you can read:
```

> 1. the official `eslint` document ["Configuring Plugins"](https://eslint.org/docs/latest/user-guide/configuring/plugins).

Reviewer: ```suggestion
1. The official `eslint` document ["Configuring Plugins"](https://eslint.org/docs/latest/user-guide/configuring/plugins).
```

2. A great explanation about the difference between an eslint `plugin` and an eslint `config` from [stackoverflow](https://stackoverflow.com/a/54522973/11554280).

> ## our plugins list

Reviewer: ```suggestion
## Our plugins
```

### eslint-plugin-migrate-to-typescript

> a compact plugin that assists to make migrate to typescript for existing porject easily, with minimum effort, prevent you from doing mistakes, focusing on the major differences between the main differences of javascript module systems to typescript ESM syntax (CommonJS and ESM).

Reviewer: ```suggestion
A compact plugin that aims to help you migrate existing javascript projects to typescript easily, with minimum effort, and minimum mistakes. Focusing on the main differences of javascript's module systems 
and typescript's ESM syntax (CommonJS and ESM).
```
I may have understood the meaning here incorrectly
